lua/fidget.lua

```diff
@@ -63,7 +64,11 @@ function base_fidget:fmt()
   self.max_line_len = #line
   for _, task in pairs(self.tasks) do
     line = options.fmt.task(task.title, task.message, task.percentage)
-    table.insert(self.lines, line)
+    if options.fmt.stack_upwards then
```

Comment:
Checking this in a loop every time `:fmt()` is called seems expensive, but this is also the kind of thing a decent optimizer will pick up immediately. Does Neovim's embedded Lua interpreter optimize this?

I suppose lifting this into a local variable is not that much less readable.

Reply:
you should consider checking a variable free.

Try checking a variable in a loop 1 million times and tell me how long it takes :)

Reply:
Awesome. I'm still new to Lua so I'm not famliar with where the usual bottlenecks are, or when the optimizer kicks in.